src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -758,6 +760,123 @@ private bool HasValidPlatformVersions()
             }
         }
 
+        internal static void AnalyzePruningResults(PackageSpec project, ILogger logger)
+        {
+            bool enablePruningWarnings =
+                SdkAnalysisLevelMinimums.IsEnabled(
+                    project.RestoreMetadata.SdkAnalysisLevel,
+                    project.RestoreMetadata.UsingMicrosoftNETSdk,
+                    SdkAnalysisLevelMinimums.PruningWarnings) &&
+                HasFrameworkNewerThanNET10(project);
+
+            if (enablePruningWarnings)
```

Comment:
I prefer the pattern of methods returning early because it reduces nesting and in my opinion is easier to read and understand.

```suggestion
            if (!enablePruningWarnings)
            {
                return false;
            }
```

Reply:
I like the nesting improvement in this case.